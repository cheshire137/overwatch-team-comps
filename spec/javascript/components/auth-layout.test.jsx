import fetchMock from 'fetch-mock'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import mockLocalStorage from '../mocks/local-storage'
import waitForRequests from '../helpers/wait-for-requests'

import AuthLayout from '../../../app/assets/javascripts/components/auth-layout.jsx'

describe('AuthLayout', () => {
  let component = null
  let store = null
  let profileReq = null

  beforeEach(() => {
    store = {
      'overwatch-team-comps': JSON.stringify({
        battletag: 'cheetos#1234',
        region: 'kr',
        platform: 'psn'
      })
    }
    mockLocalStorage(store)

    const lootboxResponse = {
      data: { avatar: 'http://example.com/image.png' }
    }
    profileReq = fetchMock.get('https://api.lootbox.eu/psn/kr/cheetos-1234/profile',
                               lootboxResponse)

    const location = { pathname: '/user' }
    component = (
      <AuthLayout location={location}><p>Hey there</p></AuthLayout>
    )
  })

  test('matches snapshot', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('saves avatar in local storage', done => {
    const expected = JSON.stringify({
      battletag: 'cheetos#1234',
      region: 'kr',
      platform: 'psn',
      avatar: 'http://example.com/image.png'
    })

    shallow(component)

    waitForRequests([profileReq], done, done.fail, () => {
      expect(store['overwatch-team-comps']).toEqual(expected)
    })
  })
})
