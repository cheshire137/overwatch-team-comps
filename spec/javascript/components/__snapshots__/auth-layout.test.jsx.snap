exports[`AuthLayout matches snapshot 1`] = `
<div
  className="layout-container">
  <div
    className="container">
    <nav
      className="nav">
      <div
        className="nav-right">
        <a
          className="nav-item is-active"
          href="/user">
          Team composition
        </a>
        <a
          className="nav-item "
          href="/user/hero-pool">
          Your hero pool
        </a>
        <button
          className="nav-item"
          onClick={[Function]}
          title="cheetos#1234"
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-user" />
        </button>
      </div>
    </nav>
  </div>
  <div
    className="layout-children-container">
    <p>
      Hey there
    </p>
  </div>
  <footer
    className="footer">
    <div
      className="container">
      <a
        href="/user/about">
        About
      </a>
      <a
        href="https://github.com/cheshire137/overwatch-team-comps"
        rel="noopener noreferrer"
        target="_blank">
        View source
      </a>
    </div>
  </footer>
</div>
`;
